quic/s2n-quic-tls/src/server.rs

```diff
@@ -30,6 +26,21 @@ impl Server {
     pub fn builder() -> Builder {
         Builder::default()
     }
+
+    /// Creates a [`Server`] from a [`ConfigLoader`]
+    ///
+    /// The caller is responsible for building the `Config`
+    /// correctly for QUIC settings. This includes:
+    /// * setting a security policy that supports TLS 1.3
+    /// * enabling QUIC support
+    /// * setting at least one application protocol
+    pub fn from_loader<L: ConfigLoader>(loader: L) -> Self {
```

Comment:
I'm surprised to see this on Server rather than on the Builder -- seems like the `keylog` bit in particular might be annoying, if it ends up being desirable to configure that *and* the loader. Mostly that's because it's a little unusual where the state for the key logging is stored directly in the Server struct. (Unlike for other configuration options where any extra state needs to be stored by the "userspace" code, not embedded into Client/Server).

Reply:
Luckily `Server` implements `ConfigLoader` so if you do want keylogging you can just nest them:

```rust
struct MyLoader {
    server: ArcSwap<Server>,
}

impl ConfigLoader for MyLoader {
    fn load(&mut self, cx: ConnectionContext) -> Config {
        self.server.load(cx)
    }
}
```

Reply:
Makes sense. A little "scary", but I think that should work - feels a little weird to pass a full new server in for the config, but I suppose there's ~no actual state involved beyond the TLS config in the not-yet-running server, so it's not actually a problem.

Reply:
That's right: the `Server` here is a server endpoint or server factory.